src/components/Header/components/UpperNav.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import { chakra, useMultiStyleConfig } from "@chakra-ui/react";
+
+import Button from "../../Button/Button";
+import Icon from "../../Icons/Icon";
+import Link from "../../Link/Link";
+import List from "../../List/List";
+
+const HeaderUpperNav = chakra(({ logInOpen, setLogInOpen }) => {
+  const styles = useMultiStyleConfig("HeaderUpperNav", {});
+  return (
+    <List
+      __css={styles.upperNav}
+      id="header-nav-upper"
+      inline
+      listItems={[
+        <>
+          <Button
+            buttonType="link"
+            className="withIcon"
+            id={logInOpen ? "closeLogInButton" : "logInButton"}
+            key={logInOpen ? "closeLogInButton" : "logInButton"}
+            onClick={() => setLogInOpen(!logInOpen)}
+          >
+            {logInOpen ? "Close" : "Log In"}
+            <Icon
+              align="right"
+              name={logInOpen ? "close" : "arrow"}
+              size="small"
+            />
+          </Button>
+          <List
+            id={logInOpen ? "logInMenuOpen" : "logInMenuClosed"}
```

Comment:
`id`s should be static, so give it only one `id` value. Hiding this should be dealt with differently.